packages/ui/src/components/EasySetup/IdentitySetIdentity.tsx

```diff
@@ -0,0 +1,127 @@
+import { Grid } from '@mui/material'
+import { styled } from '@mui/material/styles'
+import { SubmittableExtrinsic } from '@polkadot/api/types'
+import { ISubmittableResult } from '@polkadot/types/types'
+import { useCallback, useEffect, useState } from 'react'
+import { useApi } from '../../contexts/ApiContext'
+import { TextFieldStyled } from '../library'
+import { useIdentity } from '../../hooks/useIdentity'
+
+interface Props {
+  className?: string
+  from: string
+  onSetExtrinsic: (ext?: SubmittableExtrinsic<'promise', ISubmittableResult>) => void
+}
+
+interface IdentityFields {
+  display: string | undefined
+  legal: string | undefined
+  web: string | undefined
+  riot: string | undefined
+  email: string | undefined
+  image: string | undefined
+  twitter: string | undefined
+}
+
+const getRawOrNone = (val: string | undefined) => {
+  return val
+    ? {
+        Raw: val
+      }
+    : { none: null }
+}
+const getExtrinsicsArgs = (identity: IdentityFields) => {
+  const { legal, display, email, image, riot, twitter, web } = identity
+
+  return {
+    additional: [],
+    display: getRawOrNone(display),
+    legal: getRawOrNone(legal),
+    web: getRawOrNone(web),
+    riot: getRawOrNone(riot),
+    email: getRawOrNone(email),
+    pgpFingerprint: null,
+    image: getRawOrNone(image),
+    twitter: getRawOrNone(twitter)
+  }
+}
+
+const IdentitySetIdentity = ({ className, onSetExtrinsic, from }: Props) => {
```

Comment:
we normally call Props by the name of the component, so it should be IdentitySetProps . Btw The name of the component is not looking right.

Reply:
This component is doing an `identity.setIdentity` call, hence the name of this component, so this isn't a mistake. But you're right that it doesn't look easy to read or doesn't follow any other component's naming. I'll change for `SetIdentity`.

> we normally call Props...

This is not true, a quick search proves that we have a lot of such `Props`. To me, this doesn't matter at all because it's not exported.

![image](https://github.com/ChainSafe/Multix/assets/33178835/07c9c183-ad84-415a-915f-ca4044f5c597)